src/test/java/org/broadinstitute/hellbender/engine/filters/CountedReadFilterUnitTest.java

```diff
@@ -0,0 +1,150 @@
+package org.broadinstitute.hellbender.engine.filters;
+
+import htsjdk.samtools.SAMFileHeader;
+import org.broadinstitute.hellbender.utils.read.ArtificialReadUtils;
+import org.broadinstitute.hellbender.utils.read.GATKRead;
+import org.testng.Assert;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+public final class CountedReadFilterUnitTest {
+
+    // Mirrors ReadFilterUnitTest
+    static final SAMFileHeader header = ArtificialReadUtils.createArtificialSamHeader(1, 1, 10);
+    static final GATKRead goodRead = ArtificialReadUtils.createArtificialRead(header, "Zuul", 0, 2,2);
+    static final GATKRead endBad = ArtificialReadUtils.createArtificialRead(header, "Peter", 0, 1, 100);
+    static final GATKRead startBad = ArtificialReadUtils.createArtificialRead(header, "Ray", 0, -1,2);
+    static final GATKRead bothBad = ArtificialReadUtils.createArtificialRead(header, "Egon", 0, -1,100);
+    static final ReadFilter startOk = r -> r.getStart() >= 1;
+    static final ReadFilter endOk = r -> r.getEnd() <= 10;
+
+    // Helper to verify post-filtering filter state
+    private void verifyFilterState(CountedReadFilter rf, boolean expected) {
+        long count = rf.getFilteredCount();
+        String rfSummary = rf.getSummaryLine();
+
+        if (expected) {
+            Assert.assertTrue(0 == count);
+            Assert.assertEquals(-1, rfSummary.indexOf("0 read(s) filtered"));
+        }
+        else {
+            Assert.assertTrue(1 == count);
```

Comment:
I think you also need some tests that verify the individual counts of multiple filters chained via and/or.